guides/components.md

```diff
@@ -113,6 +114,35 @@ andThen(function() {
 page.modal.clickOn("I'm sure");
 ```
 
+## Functions and Getters
+
+You can use native syntax for functions and getters on pages as well to create
+methods and properties with custom functionality:
+
+```js
+import { find, click } from 'ember-test-helpers';
+
+const page = create({
+  form: {
+    scope: '.awesome-form',
+
+    firstName: fillable('#firstName'),
+    lastName: fillable('#lastName'),
+    submit: clickable('button')
+
+    login(data) {
+      return this.username(data.username)
+        .password(data.password)
+        .submit();
+    }
+
+    get role() {
+      return find(this.scope).getAttribute('aria-role');
```

Comment:
This would not always be valid, but I can't think of another example for a getter that doesn't use `findElement`. Open to suggestions

Reply:
`findElement`/`findElementWithAssert` is a public API provided to find DOM elements in the context aware fashion(it uses a proper "find" function based on the currect execution context). I think we can safely use it.

Well there is actually an issue that `findElement` currently returns a `jQuery` selection object.. But that's unrelated to the PR.

Reply:
Most of my getters compute values from existing page object properties, like this:
```js
get isFormEmpty() {
  return !this.firstName && !this.lastName;
}
```